### PR TITLE
Support ignore_keywords flag for word delimiter graph token filter

### PR DESCRIPTION
**Pull request by malpani**
This commit allows customizing the word delimiter token filters to skip processing tokens tagged as keyword through the `ignore_keywords` flag Lucene's WordDelimiterGraphFilter already exposes.

Fix for #59491

### CODE REVIEW

**Comment by romseygeek** [comment]
@elasticmachine ok to test

**Comment by malpani** [comment]
thanks for running this through the tests, the failures look unrelated
* BWC test failed on  `Building 7.9.0 didn't generate expected file /dev/shm/elastic+elasticsearch+pull-request-bwc/distribution/bwc/minor/build/bwc/checkout-7.x/distribution/archives/oss-linux-tar/build/distributions/elasticsearch-oss-7.9.0-SNAPSHOT-linux-x86_64.tar.gz`
* Default-distro failed on similar lines `Caused by: org.gradle.api.InvalidUserDataException: Building 7.9.0 didn't generate expected file /dev/shm/elastic+elasticsearch+pull-request+default-distro/distribution/bwc/minor/build/bwc/checkout-7.x/distribution/archives/linux-tar/build/distributions/elasticsearch-7.9.0-SNAPSHOT-linux-x86_64.tar.gz`

**Comment by romseygeek** [comment]
Yes, we've had an internal version bump so all the BWC tests are expecting different versions.  Can you merge in the master branch and push again?

This looks good, I think we also need to update the relevant docs (see word-delimiter-graph-filter.asciidoc for where to make changes).

**Comment by malpani** [comment]
I have added docs (reused wording from [lucene docs](https://lucene.apache.org/core/8_6_0/analyzers-common/org/apache/lucene/analysis/miscellaneous/WordDelimiterGraphFilter.html#IGNORE_KEYWORDS)) and also merged in the latest upstream/master

**Review by romseygeek — APPROVED** [review]
LGTM, thanks @malpani!  @jrodewig if you're happy with the docs change I'll merge.

File: docs/reference/analysis/tokenfilters/word-delimiter-graph-tokenfilter.asciidoc [hunk 1]
```diff
@@ -270,6 +270,12 @@ If `true`, the filter includes tokens consisting of only alphabetical characters
 in the output. If `false`, the filter excludes these tokens from the output.
 Defaults to `true`.
 
+`ignore_keywords`::
+(Optional, boolean)
+If `true`, the filter suppresses processing tokens with
+{lucene-core-javadoc}/org/apache/lucene/analysis/tokenattributes/KeywordAttribute.html#isKeyword()[KeywordAttribute.isKeyword()=true]
```
Comment (jrodewig) [inline docs/reference/analysis/tokenfilters/word-delimiter-graph-tokenfilter.asciidoc:276]:
```suggestion
If `true`, the filter suppresses processing tokens with
a `keyword` attribute of `true`.
```

**Review by jrodewig — APPROVED** [review]
LGTM. Made a small change to the docs.

**Comment by romseygeek** [comment]
Thanks @malpani !

**Comment by malpani** [comment]
thanks @romseygeek for reviewing and merging this in!

**Pull request merged**